src/generic/AbsMSeries.jl

```diff
@@ -0,0 +1,774 @@
+###############################################################################
+
+# set the exponent vector of the underlying polynomial (used internally)
+function set_exponent_vector!(a::AbsMSeries, i::Int64, v::Vector{Int64})
+    a.poly = set_exponent_vector!(poly(a), i, v)
+    return a
+end
+
+# set the coefficient of the underlying polynomial (used internally)
+function setcoeff!(a::AbsMSeries{T}, i::Int64, c::T) where T <: RingElement
+    a.poly = setcoeff!(poly(a), i, c)
+    return a
+end
+
+# used by MPolyBuildCtx by evaluation
+function sort_terms!(a::AbsMSeries)
+    a.poly = sort_terms!(poly(a))
+    return a
+end
+
+# used by MPolyBuildCtx by evaluation
+function combine_like_terms!(a::AbsMSeries)
+    a.poly = combine_like_terms!(poly(a))
+    return a
+end
+
+###############################################################################
+#
+#   AbstractString I/O
+#
+###############################################################################
+
+function AbstractAlgebra.expressify(a::AbstractAlgebra.AbsMSeriesElem,
+                                        x = vars(parent(a)); context = nothing)
+    apoly = poly(a)
+
+    poly_sum = Expr(:call, :+)
+    n = nvars(parent(apoly))
+    
+    iter = zip(coeffs(apoly), exponent_vectors(apoly))
+    cv = reverse(collect(iter))
+    
+    for (c, v) in cv
+        prod = Expr(:call, :*)
+        if !isone(c)
+            push!(prod.args, expressify(c, context = context))
+        end
+        for i in n:-1:1
+            if v[i] > 1
+                push!(prod.args, Expr(:call, :^, x[i], v[i]))
+            elseif v[i] == 1
+                push!(prod.args, x[i])
+            end
+        end
+        push!(poly_sum.args, prod)
+    end
+    
+    sum = Expr(:call, :+)
+
+    push!(sum.args, poly_sum)
+
+    for i in nvars(parent(a)):-1:1
+        push!(sum.args, Expr(:call, :O, Expr(:call, :^, x[i], a.prec[i])))
+    end
+
+    return sum
+end
+
+function Base.show(io::IO, a::AbstractAlgebra.MSeriesElem)
+    print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+  
+function Base.show(io::IO, ::MIME"text/plain", a::AbstractAlgebra.MSeriesElem)
+    print(io, AbstractAlgebra.obj_to_string(a, context = io))
+end
+  
+function show(io::IO, a::AbstractAlgebra.MSeriesRing)
+    v = join([String(s) for s in vars(a)], ", ")
+    print(io, "Multivariate power series ring in ", v, " over ")
+    print(IOContext(io, :compact => true), base_ring(a))
+end
+
+###############################################################################
+#
+#   Exponent helper functions (not exported)
+#
+###############################################################################
+
+function exponents_clamp_zero_to_prec(a::Vector{Int}, prec::Vector{Int})
+    return [a[i] == 0 ? prec[i] : a[i] for i in 1:length(a)]
+end
+
+function exponents_lt(v::Vector{Int}, p::Vector{Int})
+    for i = 1:length(v)
+        if v[i] >= p[i]
+            return false
+        end
+    end
+    return true
+end
```

Comment:
Just for fun, if you like functional one-liners:
```suggestion
exponents_lt(v::Vector{Int}, p::Vector{Int}) = all(((x, y),) -> x < y, zip(v, p))
```

Reply:
Nice. I like this better.